lib/awful/hotkeys_popup/widget.lua

```diff
@@ -227,7 +227,7 @@ function widget.new()
         local max_height_px = height - group_label_height
         local column_layouts = {}
         for _, group in ipairs(available_groups) do
-            local keys = cached_awful_keys[group] or widget_instance.additional_hotkeys[group]
+            local keys = awful.util.table.join(cached_awful_keys[group], widget_instance.additional_hotkeys[group])
```

Comment:
If possible, avoid the `awful.util.table.join`. It's slow and unnecessary. Better add a second level to the 2 for loops that use it.